Sources/EpoxyCollectionView/CollectionView/ReusableViews/CollectionViewCell.swift

```diff
@@ -61,73 +61,108 @@ public final class CollectionViewCell: UICollectionViewCell, ItemCellView {
+      if
+        verticalFittingPriority == .required &&
+        contentView.bounds.height != layoutAttributes.size.height
+      {
+        contentView.bounds.size.height = layoutAttributes.size.height
+      }
 
-    // In some cases, `contentView`'s required width and height constraints
-    // (created from its auto-resizing mask) will not have the correct constants before invoking
-    // `systemLayoutSizeFitting(...)`, causing the cell to size incorrectly. This seems to be a
-    // UIKit bug.
-    // https://openradar.appspot.com/radar?id=5025850143539200
-    // The issue seems most common when the collection view's bounds change (on rotation).
-    // We correct for this by updating `contentView.bounds`, which updates the constants used by the
-    // width and height constraints created by the `contentView`'s auto-resizing mask.
+      let size: CGSize
+      if horizontalFittingPriority != .required || verticalFittingPriority != .required {
+        // Self-sizing is required in at least one dimension.
+        size = super.systemLayoutSizeFitting(
+          layoutAttributes.size,
+          withHorizontalFittingPriority: horizontalFittingPriority,
+          verticalFittingPriority: verticalFittingPriority)
+      } else {
+        // No self-sizing is required; respect whatever size the layout determined.
+        size = layoutAttributes.size
+      }
 
-    if
-      horizontalFittingPriority == .required &&
-      contentView.bounds.width != layoutAttributes.size.width
-    {
-      contentView.bounds.size.width = layoutAttributes.size.width
+      layoutAttributes.size = size
+      preferredAttributes = layoutAttributes
+    } else {
+      preferredAttributes = super.preferredLayoutAttributesFitting(layoutAttributes)
     }
 
+    // On iOS 15, cells that return an unstable size can result in a layout recursion crash. The
+    // root cause of these crashes is likely an ambiguous layout due to misconfigured constraints.
+    // Unfortunately, finding each and every one of these ambiguous layouts can be challenging. This
+    // code works around this new behavior / crash by giving up on self-sizing after 5 attempts.
+    // Once a component has been sized differently 5 times, we assume that it has an ambiguous
+    // layout and we simply return the last computed size for the remainder of the component's life.
+    // We reset the count on cell reuse and when the cell is reconfigured in `CollectionView`'s
+    // `configure` function.
     if
-      verticalFittingPriority == .required &&
-      contentView.bounds.height != layoutAttributes.size.height
+      CollectionViewConfiguration.shared.enableLayoutRecursionWorkaround,
+      let previousComputedSize = previousComputedSize,
+      numberOfNewComputedSizes >= 5
     {
-      contentView.bounds.size.height = layoutAttributes.size.height
+      preferredAttributes.size = previousComputedSize
```

Comment:
Could we trigger an assertion with the epoxy logger at this point that includes a description of `view`? That way we should be able to track down in which components this is happening since these assertions can flow into our logging system

Reply:
Great idea - adding!